File: domains/system/service/comment.go
package service

import (
	"context"

	"github.com/ayonli/bilingo/common"
	"github.com/ayonli/bilingo/domains/system/models"
	"github.com/ayonli/bilingo/domains/system/repo"
	"github.com/ayonli/bilingo/domains/system/types"
)

func GetComment(ctx context.Context, id uint) (*models.Comment, error) {
	return repo.CommentRepo.Get(ctx, id)
}

func ListComments(ctx context.Context, query types.CommentListQuery) (*common.PaginatedResult[models.Comment], error) {
	return repo.CommentRepo.List(ctx, &query)
}

func CreateComment(ctx context.Context, data *types.CommentCreate) (*models.Comment, error) {
	return repo.CommentRepo.Create(ctx, data)
}

func UpdateComment(ctx context.Context, id uint, updates *types.CommentUpdate) (*models.Comment, error) {
	return repo.CommentRepo.Update(ctx, id, updates)
}

func DeleteComment(ctx context.Context, id uint) error {
	return repo.CommentRepo.Delete(ctx, id)
}

File: domains/comment/api/comment.ts
import type { ApiResponse, PaginatedResult } from "@/common"
import { ApiEntry } from "@/client"
import type { Comment } from "../models"
import type { CommentCreate, CommentListQuery, CommentUpdate } from "../types"

const commentApi = new ApiEntry("/comments")

export async function getComment(id: number): ApiResponse<Comment> {
    return await commentApi.get("/" + id)
}

export async function listComments(
    query: Partial<CommentListQuery>,
): ApiResponse<PaginatedResult<Comment>> {
    return await commentApi.get("/", query)
}

export async function createComment(data: CommentCreate): ApiResponse<Comment> {
    return await commentApi.post("/", null, data)
}

export async function updateComment(id: number, data: CommentUpdate): ApiResponse<Comment> {
    return await commentApi.patch("/" + id, null, data)
}

export async function deleteComment(id: number): ApiResponse<null> {
    return await commentApi.delete("/" + id)
}
